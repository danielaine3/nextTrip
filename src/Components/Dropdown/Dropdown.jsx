import React from 'react'

export const Dropdown = ({
  id,
  options, 
  onSelect,
  selected,
  title,
}) => {
  const handleChange = (event) => {
    onSelect(event.target.value)
  }
  console.log('options', options)


  return (
    <>
      <label id={`${id}-label`}>{title}</label>
      <select
        id={id}
        onChange={handleChange}
        value={selected}
      >
        {options.map(option => {
          return (
            <option
              key={option.route_id}
              value={option.route_id}
            >
              {option.route_label}
            </option>
          )
        })}
      </select>
    </>
  )
}
