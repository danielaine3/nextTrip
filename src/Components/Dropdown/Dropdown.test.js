import React from 'react'
import { render } from '@testing-library/react'
import { Dropdown } from './Dropdown'

const options = [
  {direction_id: '0', direction_name: 'Northbound'},
  {direction_id: '1', direction_name: 'Southbound'},
]
describe('<Dropdown />', () => {
  it('should render Dropdown', async () => {
    const { asFragment} = await render(
      <Dropdown
        id={'id'}
        options={options}
        onSelect={jest.fn()}
        selected={''}
        title={'title'}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})