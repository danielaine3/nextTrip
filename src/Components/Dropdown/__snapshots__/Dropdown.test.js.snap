// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should render Dropdown 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root makeStyles-formControl-1"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink="false"
      id="id-label"
    >
      title
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="id-label id"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
        id="id"
        role="button"
        tabindex="0"
      >
        <span>
          â€‹
        </span>
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-4"
        >
          <span>
            title
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
